shared/charts/backup-exporter/templates/configmap.yaml

```diff
@@ -11,17 +11,24 @@ metadata:
 data:
   backup.sh: |
     set -e
+    DESTINATION_BUCKET={{ .Values.destinationBucket }}
+    INTERMEDIATE_BUCKET={{ .Values.localIntermediateBucket }}
+    LOG_BUCKET={{ .Values.logBucket }}
+
+    gsutil -q ls gs://${DESTINATION_BUCKET}/ 2>1 >/dev/null || { echo "Destination bucket is not accessible"; exit 1; }
+    gsutil -q ls gs://${INTERMEDIATE_BUCKET}/ 2>1 >/dev/null || { echo "Intermediate bucket is not accessible"; exit 1; }
+    gsutil -q ls gs://${LOG_BUCKET}/ 2>1 >/dev/null || { echo "Log bucket is not accessible"; exit 1; }
+
     gcloud config set account {{ .Values.serviceAccountName }}
     SNAPSHOTS=$(for i in `seq 0 $(({{ .Values.replicaCount }}-1))`; do gcloud compute snapshots list --sort-by=~creationTimestamp,STATUS --limit=1 --format="value[separator=';'](name,status)" --filter="name~'pv-database-storage-couchdb-couchdb-$i-*'" | cut -f1 -d\; ; done)
-    DESTINATION_BUCKET={{ .Values.destinationBucket }}
     TIMESTAMP=$(date +%F_%H%M%S%Z)
     for snapshot in ${SNAPSHOTS}; do
       IMAGE_FOUND=$(gcloud compute images list --format="value[separator=';'](name)" --filter=name=image-disk-${snapshot})
       if [ "${IMAGE_FOUND}" == "image-disk-${snapshot}" ]; then
         gcloud -q compute images delete image-disk-${snapshot}
       fi
       gcloud -q compute images create image-disk-${snapshot} --source-snapshot ${snapshot}
-      gcloud -q --verbosity=debug compute images export --log-location={{ .Values.logBucket }} --destination-uri gs://{{ .Values.localIntermediateBucket }}/${TIMESTAMP}-${snapshot}.tar.gz --image image-disk-${snapshot}
+      gcloud -q --verbosity=debug compute images export --log-location=${LOG_BUCKET} --destination-uri gs://${INTERMEDIATE_BUCKET}/${TIMESTAMP}-${snapshot}.tar.gz --image image-disk-${snapshot}
       gcloud -q compute images delete image-disk-${snapshot}
-      gsutil -q mv gs://{{ .Values.localIntermediateBucket }}/${TIMESTAMP}-${snapshot}.tar.gz ${DESTINATION_BUCKET}/${TIMESTAMP}-${snapshot}.tar.gz
+      gsutil -q mv gs://${INTERMEDIATE_BUCKET}/${TIMESTAMP}-${snapshot}.tar.gz ${DESTINATION_BUCKET}/${TIMESTAMP}-${snapshot}.tar.gz
```

Comment:
Might be a good idea to put quotes around these user-provided values.

Reply:
I quoted all the strings name that use external variables.